### ISSUE TITLE
build failure with latest nightly (lifetime mismatch)

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang-nursery/rust-semverver#70]
`rustc 1.31.0-nightly (8c4ad4e9e 2018-10-04)`

````
   Compiling cargo v0.29.0
   Compiling semverver v0.1.18
error[E0623]: lifetime mismatch
   --> /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/semverver-0.1.18/src/semcheck/traverse.rs:329:25
    |
325 | fn diff_fn(changes: &mut ChangeSet, tcx: TyCtxt, old: Def, new: Def) {
    |                                          ------
    |                                          |
    |                                          this type is declared with multiple lifetimes...
...
329 |     let old_const = tcx.is_const_fn(old_def_id);
    |                         ^^^^^^^^^^^ ...but data with one lifetime flows into the other here

error[E0623]: lifetime mismatch
   --> /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/semverver-0.1.18/src/semcheck/traverse.rs:330:25
    |
325 | fn diff_fn(changes: &mut ChangeSet, tcx: TyCtxt, old: Def, new: Def) {
    |                                          ------
    |                                          |
    |                                          this type is declared with multiple lifetimes...
...
330 |     let new_const = tcx.is_const_fn(new_def_id);
    |                         ^^^^^^^^^^^ ...but data with one lifetime flows into the other here

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0623`.
error: failed to compile `semverver v0.1.18`, intermediate artifacts can be found at `/tmp/cargo-install7N9N6d`

Caused by:
  Could not compile `semverver`.
````

**Comment by ibabushkin** [comment]
Heh, you seem to be consistently quicker than my CI cronjob :)

**Comment by ibabushkin** [comment]
Fixed in 0.1.19 on crates.io.
